engine/runtime-compiler/src/main/java/org/enso/compiler/pass/MiniIRPass.java

```diff
@@ -103,10 +105,17 @@ public String toString() {
    * Combines two mini IR passes into one that delegates to both of them.
    *
    * @param first first mini pass (can be {@code null})
-   * @param second second mini pass
-   * @return a combined pass that calls both non-{@code null} of the provided passes
+   * @param second second mini pass (can be {@code null})
+   * @return a combined pass that calls both non-{@code null} of the provided passes. {@code null}
+   *     if both provided passes are {@code null}.
    */
   public static MiniIRPass combine(MiniIRPass first, MiniIRPass second) {
+    if (first == null) {
+      return second;
+    }
+    if (second == null) {
+      return first;
+    }
```

Comment:
Oh, the check for `null` is here. I was expecting it to be inside of `static MiniIRPass chain` - that was the whole point of having a `factory` method - otherwise one can use just the constructor directly, but one has to use `combine` method consistently wherever allocating new chain then.

Reply:
Check for null moved to the factory method in 273bea8f37c724ffa9ae424af0b2314707af4ac4